### PR TITLE
Global Help options

### PR DESCRIPTION
**Pull request by ovinis**
#### About
This PR contains updates to allow users to use the following commands to get help on the functionality available on Tabcmd2:

```
tabcmd2 help             -- Help for tabcmd commands 
tabcmd2 help <a command> -- Show Help for a specific command
tabcmd2 help commands    -- List all available commands
```

### CODE REVIEW

File: pythontabcmd2/parsers/create_project_parser.py [hunk 1]
```diff
@@ -29,6 +31,7 @@ def create_project_parser():
                                            help='content permission ')
 
         args = create_project_parser.parse_args(sys.argv[2:])
+        print("THESE ARE THE SYS ARGV", sys.argv)
```
Comment (jacalata) [inline pythontabcmd2/parsers/create_project_parser.py:34]: heh. debugging?

**Review by jacalata — COMMENTED** [review]
Looks good. If I were writing a new command, it looks like I'd just have to edit MyCommandParser, map_of_commands, and map_of_parsers to get the complete user help and options set up?

if replacing the arg[index] values with a const would be a lot of work, it's fine to leave that for another changelist.

File: pythontabcmd2/parsers/export_parser.py [hunk 2]
```diff
@@ -44,7 +46,10 @@ def export_parser():
         export_parser.add_argument('--filter', '-vf', metavar='COLUMN:VALUE',
                                    help='View filter to apply to the view')
         args = export_parser.parse_args(sys.argv[3:])
-        url = sys.argv[2]
+        try:
+            url = sys.argv[2]
+        except Exception as ex:
+            print(ex)
```
[the remark this answers is not in the document]
Comment (ovinis) [reply pythontabcmd2/parsers/export_parser.py]: Thanks for catching it. Fixed in latest commit to ask the user to enter a valid URL

File: pythontabcmd2/parsers/reencrypt_parser.py [hunk 3]
```diff
@@ -8,15 +8,20 @@ class ReencryptExtractsParser:
...
         """Method to parse reencrypt extracts arguments passed by the user"""
+        site_name = ""
         parent_parser = ParentParser()
         parser = parent_parser.parent_parser_with_global_options()
         subparsers = parser.add_subparsers()
         reencrypt_extract_parser = subparsers.add_parser('reencryptextracys',
                                                          parents=[parser])
-        site_name = sys.argv[2]
+        try:
+            site_name = sys.argv[2]
```
[the remark this answers is not in the document]
Comment (ovinis) [reply pythontabcmd2/parsers/reencrypt_parser.py]: Fixed all the magic numbers in all parsers. Some parsers use the same arg index, some parsers don't. For example, 
Adding users to a group `tabcmd addusers "Development" --users "users.csv"` we don't have a flag when the user specifies a group name. Whereas in the createproject command `tabcmd createproject -n "Quarterly_Reports"` user has to specify the name flag.  In summary, when taking user input the indices vary.

**Comment by ovinis** [comment]
@jacalata Yes, If you are writing a new command you'd have to edit MyCommandParser, map_of_commands, and map_of_parsers to get the complete user help and options set up.

**Pull request merged**